src/Compilers/CSharp/Test/Symbol/Compilation/GetSemanticInfoTests.cs

```diff
@@ -918,8 +918,8 @@ public void EnumOffsets()
             // long
             EnumOffset(ConstantValue.Create((long)long.MinValue), 1, EnumOverflowKind.NoOverflow, ConstantValue.Create((long)(long.MinValue + 1)));
             EnumOffset(ConstantValue.Create((long)long.MinValue), 2, EnumOverflowKind.NoOverflow, ConstantValue.Create((long)(long.MinValue + 2)));
-            EnumOffset(ConstantValue.Create((long)-2), 1, EnumOverflowKind.NoOverflow, ConstantValue.Create((long)(-1)));
-            EnumOffset(ConstantValue.Create((long)-2), 3, EnumOverflowKind.NoOverflow, ConstantValue.Create((long)(1)));
+            EnumOffset(ConstantValue.Create((long)-2), 1, EnumOverflowKind.NoOverflow, ConstantValue.Create((long)-1));
```

Comment:
Mumble mumble something about precedence... lets see is `(X)-1` the same as `(X)-(1)` or `(X)(-1)`? Does that even apply here? Who cares, just add parens until it is crystal clear.

Reply:
Generally speaking I have sympathy for parens on the right-hand-side of a cast even if not strictly needed.

Reply:
Even `(T)(x)`

Reply:
this one is one i feel is safe to let the user know about by default. Note that we are not actually removing these for users, just subtly letting them know they don't actually need them.  In a sense, the feature acts to actually answer the question for people whose minds do go ```lets see is (X)-1 the same as (X)-(1) or (X)(-1)...``` When you start adding the extra parens, you can see "oh, nope... it's not actually necessary.  Cool, i'll just leave them out".

Reply:
The fact that the code author was unsure should be a hint that this will be a problem for the reader too. So leave them in!

Reply:
@gafter are you saying that solely for *casts* we should have that logic?  or we shoudl apply that rule in general?  If we applied it in general that would mean, by default, we would not suggest removing any parens anywhere.  That's a totally fine position to take, i just want to know if that's what you're suggesting.

Note: if we take that approach, then we would not remove in cases like ```if ((x > y))```.  However, i'm not sure how we'd be able to tell that wasn't just another case of "The fact that the code author was unsure should be a hint that this will be a problem for the reader too. So leave them in!"

Thoughts?

Reply:
I think casts in particular can be more clear with parens around the target.

Reply:
Ok.  Importantly:

This PR is not intended to go in.  It's just to help us understand what, by default, the IDE would be pointing out to users.


it is someone interesting that in a single line, with the same sort of construct, parens were used inconsistently.  i.e.:

```((long)-2), 1, EnumOverflowKind.NoOverflow, ConstantValue.Create((long)(-1)))```

Here, the ```-2``` is *not* surrounded by parens, but the ```-1``` is.  We'll discuss if we think unnecessary parens around casts are things that should be marked by default as something that can be removed, or if we think it should be something the user has to opt-into wanting the fading for.

Reply:
@gafter i'm continuing this conversation over at: https://github.com/dotnet/roslyn/pull/21316#issuecomment-374060282